picard/ui/theme.py

```diff
@@ -81,12 +81,24 @@ def get_syntax_theme(self):
     def update_palette(self, palette, dark_theme, accent_color):
         if accent_color:
             accent_text_color = QtCore.Qt.white if accent_color.lightness() < 160 else QtCore.Qt.black
-            palette.setColor(QtGui.QPalette.Highlight, accent_color)
-            palette.setColor(QtGui.QPalette.HighlightedText, accent_text_color)
+            palette.setColor(QtGui.QPalette.Active, QtGui.QPalette.HighlightedText, accent_text_color)
+
+            if accent_text_color == QtCore.Qt.white:
+                contrasting_accent_color = accent_color.lighter()
+            else:
+                contrasting_accent_color = accent_color.darker()
+
+            contrasting_accent_text_color = QtCore.Qt.white if contrasting_accent_color.lightness() < 160 else QtCore.Qt.black
+            palette.setColor(QtGui.QPalette.Inactive, QtGui.QPalette.HighlightedText, contrasting_accent_text_color)
```

Comment:
I think we should set the accent color only for the active state here and don't set anything for inactive. For the white theme on Windows it will already use a gray color for the inactive state, so it matches previous behavior (colored active highlight, grey inactive highlight).

When using a dark theme we should instead set proper colors for the inactive highlight in `WindowsTheme.update_palette`, something like this would do:

```
        def update_palette(self, palette, dark_theme, accent_color):
            # Adapt to Windows 10 color scheme (dark / light theme and accent color)
            super().update_palette(palette, dark_theme, accent_color)
            if dark_theme:
                # ...
                palette.setColor(QtGui.QPalette.Inactive, QtGui.QPalette.Highlight, QtGui.QColor(51, 51, 51))
                palette.setColor(QtGui.QPalette.Inactive, QtGui.QPalette.HighlightedText, QtCore.Qt.white)
```

Reply:
Not sure on this one. My default accent is dark grey (63, 63, 63) and contrast is really poor by assuming (51, 51, 51). First two columns are selecting the album on the right panel and the artist field assuming (51,51,51). Last two columns are using the code in the PR, which results in active highlight (94,94,94) while inactive is kept at (63,63,63). I told you that the accent thing was a problem =x
![Untitled](https://user-images.githubusercontent.com/1678454/87336334-1034bc80-c518-11ea-9b72-c201be71b2a3.png)

Reply:
> Maybe do the same thing, get the nearest gray and use it instead of assuming 51,51,51?

Not sure, what would be the nearest grey for your (94,94,94)? If we choose the nearest grey for the selected accent color and  make that lighter it could work.

I chose (51,51,51) because this matches what Windows uses. And Windows seems to use the same grey for inactive independent of the accent color. Not sure why we should make things more complicated then Windows is doing it. E.g. in Explorer active / inactive selection look like on your screenshot if I choose a dark accent color.

![grafik](https://user-images.githubusercontent.com/29852/87338918-25c0db00-c546-11ea-9edf-fb01ae9e4de0.png)

Reply:
Getting the nearest gray to the "contrasting" color. It isn't as good as I thought.
![image](https://user-images.githubusercontent.com/1678454/87339333-d5815300-c51c-11ea-95fd-e05d99666b3c.png)

![image](https://user-images.githubusercontent.com/1678454/87339682-580a1280-c51d-11ea-83b7-84a77b0a83a3.png)

Reply:
Using the perceived brightness. Far less aggressive.

![image](https://user-images.githubusercontent.com/1678454/87340960-3578f900-c51f-11ea-9157-600132d0e0c0.png)


![image](https://user-images.githubusercontent.com/1678454/87340749-e337d800-c51e-11ea-8f3a-23f2b7f52c42.png)

Reply:
> Not sure, what would be the nearest grey for your (94,94,94)? If we choose the nearest grey for the selected accent color and make that lighter it could work.

The nearest gray for a gray color is itself. For other colors it is more complicated.

> I chose (51,51,51) because this matches what Windows uses. And Windows seems to use the same grey for inactive independent of the accent color. Not sure why we should make things more complicated then Windows is doing it. E.g. in Explorer active / inactive selection look like on your screenshot if I choose a dark accent color.

Hmm, weird. I don't know then.

Reply:
I think we could go with the perceived brightness variant. I any case we should keep it grey, that's matching both the Windows behavior and the behavior without overriding the accent color.

But IMHO using a fixed gray for inactive would be ok. It is what Windows does, and it is also what happens on macOS. There Qt automatically uses the system configured accent color for active selections, and a fixed gray for inactive selections:

![grafik](https://user-images.githubusercontent.com/29852/87457776-3c7c3600-c609-11ea-94e2-7cdce8cb66f0.png)

Mimicking on Windows what Qt is doing automatically on macOS was kind of my idea why I chose to use the accent color.

Reply:
Well, reverted some changes to use the fixed color. The perceived brightness version is on a different branch. If anyone complain, it is there. :)

Reply:
Ok, let's merge this. If needed we can tweak it, but I think this is now working well enough for most use cases.